app/src/main/java/com/babylon/wallet/android/domain/usecases/ResolveDAppInTransactionUseCase.kt

```diff
@@ -31,9 +31,20 @@ class ResolveDAppInTransactionUseCase @Inject constructor(
             ).mapCatching { dApps ->
                 val dApp = dApps.first()
                 dApp to dApp.claimedEntities.contains(componentAddress)
+                DAppInTransaction(
+                    dApp = dApp,
+                    isVerified = dApp.claimedEntities.contains(componentAddress),
+                    componentAddresses = dApp.claimedEntities.filter { it.startsWith("component_") }
+                )
             }
         } else {
             Result.failure(RadixWalletException.DappVerificationException.WrongAccountType)
         }
     }
 }
+
+data class DAppInTransaction(
+    val dApp: DApp,
+    val isVerified: Boolean,
+    val componentAddresses: List<String>
```

Comment:
This shouldn't be a val itself. It can be a backing field, since all of its info can be derived from `dApp`:

```
val componentAddresses: List<String>
    get() = dApp.claimedEntities.filter { it.startsWith("component_") }
    
// or you can use lazy { } instead of get() so as to compute this only once.    
```
Whenever we can find some information from a property we don't need to attach that information as a separate property. This can lead to bugs such as someone inserting a dapp to this object but forgetting to add the correct component addresses to this object such as that `componentAddresses != dApp.claimedEntities.filter { it.startsWith("component_") }`

Reply:
actually why not to move the `componentAddresses` inside the `DApp` and get rid of the extra data class `DAppInTransaction` ?

Reply:
actually why not to move the `componentAddresses` inside the `DApp` data class and get rid of the extra data class `DAppInTransaction`?

Reply:
> componentAddresses

@giannis-rdx Yes we could move `componentAddresses` inside `dApp` data class however we cannot move there isVerified as it depends on external `componentAddress`,  which is why we cannot remove DAppInTransaction.

Reply:
yes my point was to keep what we had before with the Pair and pass the componentAddresses inside the `DApp`. 
In any case it is not a hard requirement from my side. 
up to you!